integrations/iron/src/main/scala/sttp/iron/codec/iron/TapirCodecIron.scala

```diff
@@ -0,0 +1,47 @@
+package sttp.tapir.codec.iron
+
+import sttp.tapir.Schema
+import io.github.iltotore.iron.Constraint
+import io.github.iltotore.iron.:|
+import io.github.iltotore.iron.refineEither
+import io.github.iltotore.iron.refineOption
+import sttp.tapir.CodecFormat
+import sttp.tapir.Codec
+import sttp.tapir.DecodeResult
+import sttp.tapir.Validator
+import sttp.tapir.ValidationResult
+import scala.reflect.ClassTag
+import sttp.tapir.ValidationError
+
+trait TapirCodecIron {
+
+  inline given [Value, Predicate: ClassTag](using
+      inline vSchema: Schema[Value],
+      inline constraint: Constraint[Value, Predicate]
+  ): Schema[Value :| Predicate] =
+    vSchema.validate(validator).map[Value :| Predicate](v => v.refineOption[Predicate])(identity)
+
+  inline given [R, V, P: ClassTag, CF <: CodecFormat](using
+      inline tm: Codec[R, V, CF],
+      inline constraint: Constraint[V, P]
+  ): Codec[R, V :| P, CF] = {
+
+    implicitly[Codec[R, V, CF]]
```

Comment:
In Scala 3, you should use `summon` instead of `implicitly`

Reply:
That's a leftover